web_widget_text_markdown/static/src/js/web_widget_text_markdown.js

```diff
@@ -42,8 +43,38 @@ odoo.define("web_widget_text_markdown.FieldTextMarkDown", function(require) {
 
         start: function() {
             this._super();
+            // This is custom how to add CSS class and styles
+            // Showdown does not support this out of the box.
+            // eslint-disable-next-line max-len
+            // Ref: https://github.com/showdownjs/showdown/wiki/Add-default-classes-for-each-HTML-element
+            const classMap = {
+                table: "table table-striped",
+            };
+
+            const styleMap = {
+                h1: "font-size: 2.5em;",
+            };
```

Comment:
I'm OK with using this method when classes are involved, for example `table-striped` is something useful and not easily added with CSS.

However if it comes to `h1` you will get code like this:

```
<h1 style="font-size: 2.5em;">Header</h1>
```

Using such inline styles is not a good practice IMO, your whole HTML code will get cluttered with inline style. For that I prefer just to use a CSS with

```
.o_field_text markdown { h1 { font-size: "2.5 em" }}
```

If it's not applying correctly, then perhaps some other style is taking precedence: in that case you can add `!important` to the end (but not if it's not necessary!)

Reply:
@thomaspaulb As mentioned, the 13.0 bootstrap css file seems to override the v8.0 css changes i introduced.
see current commit for my implementation maybe i missed something.

But from my investigation the 13.0 bootstrap css uses directly the elements (h1, pre, table etc) so there aren't classes i can
override easily unless i also override directly which in this case affects Odoo. I am not sure if its cache, or hierarchy of adding assets maybe our css comes later after bootstrap. still investigating on the theories.

Also maybe we can settle for how 13.0 wrks without adding the v8.0 css changes since also it was not included in v11.0 which I began my port from. It might allow flexibility for users to choose their own theming styles, IMO.

Reply:
@KKamaa Could it be the name of the class?

The widget has `.oe-*`:

![image](https://user-images.githubusercontent.com/1466356/126558276-b154f240-f386-4039-9bd2-cfc2b85516ed.png)

Whereas the CSS has `.o-*`:

![image](https://user-images.githubusercontent.com/1466356/126558339-38b4c57e-861d-4233-b99f-35b41dfcc99b.png)

Reply:
People are flexible to override this in their own way, but we need to provide a good default.